### PR TITLE
Number service

### PR DESCRIPTION
**Pull request by maryesima**
adding number service

### CODE REVIEW

File: initial/src/main/java/com/example/resources/CalculatorResource.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Random;
+import java.util.List;
+
+@RestController
+public class CalculatorResource {
+    public static final String URL_PREFIX="/calculator";
+
+	@RequestMapping(URL_PREFIX + "/random")
```
Comment (etwei-linkedin) [inline initial/src/main/java/com/example/resources/CalculatorResource.java:14]: Not a blocker. But going forward, can you add *javadoc* for these public methods?

File: initial/src/main/java/com/example/resources/NumbersResource.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class NumbersResource {
+    public static final String URL_PREFIX="/numbers";
+
+    @RequestMapping(URL_PREFIX + "/random")
+    public int getRandom() {
+        NumberService ns = new NumberService();
```
Comment (etwei-linkedin) [inline initial/src/main/java/com/example/resources/NumbersResource.java:16]: NOT AN ISSUE, this comment is for my own note. We will soon go through @Component and make sure all service classes are initiated during service startup.

File: initial/src/main/java/com/example/services/NumberService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.Random;
+
+@RestController
+public class NumberService {
+    public static final Random RAND = new Random();
+
+    /**
+     *
+     * @return
+     */
```
Comment (etwei-linkedin) [inline initial/src/main/java/com/example/services/NumberService.java:13]: Ah oh, *java doc* :)

File: initial/src/test/java/com/example/application/CalculatorControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        Assertions.assertEquals(10, CalculatorService.applyOp("/", 20, 2));
+    }
+
+    @Test
+    public void testApplyOpDivideAndRound() {
+        Assertions.assertEquals(7, CalculatorService.applyOp("/", 15, 2));
+    }
+
+    @Test
+    public void testApplyOpDivideByZero() {
```
Comment (etwei-linkedin) [inline initial/src/test/java/com/example/application/CalculatorControllerTests.java:41]: A good name for test method is always like *test{testing_scenario_name}_expects{results}*, e.g. testApplyOpDivideByZero_expectsUnsupportedOperationException. Don't be afraid of the length of the testing method name.

File: initial/src/main/java/com/example/resources/CalculatorResource.java (hunk 1 above)
Comment (maryesima) [reply initial/src/main/java/com/example/resources/CalculatorResource.java:14]: added!

**Review by etwei-linkedin — APPROVED** [review]

**Pull request merged**